backend/services/courses/term.py

```diff
@@ -0,0 +1,159 @@
+        """
+        # Select all entries in `Organization` table
+        query = select(TermEntity)
+
+        entities = self._session.scalars(query).all()
+
+        # Convert entries to a model and return
+        return [entity.to_details_model() for entity in entities]
+
+    def get_from_id(self, id: str) -> TermDetails:
+        """Gets the term from the table for an id.
+
+        Returns:
+            TermDetails: Term based on the id.
+        """
+        # Select all entries in the `Term` table and sort by end date
+        query = select(TermEntity).filter(TermEntity.id == id)
+        entity = self._session.scalars(query).one_or_none()
+
+        # Raise an error if no entity was found.
+        if entity is None:
+            raise ResourceNotFoundException(f"Term with id: {id} does not exist.")
+
+        # Return the model
+        return entity.to_details_model()
+
+    def create(self, subject: User, term: Term) -> Term:
+        """Creates a new term.
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            term (Term): Term to add to table
+
+        Returns:
+            Term: Object added to table
+        """
+
+        # Check if user has admin permissions
+        self._permission.enforce(subject, "course.admin", f"term")
+
+        # Create new object
+        term_entity = TermEntity.from_model(term)
+
+        # Add new object to table and commit changes
+        self._session.add(term_entity)
+        self._session.commit()
+
+        # Return added object
+        return term_entity.to_model()
+
+    def update(self, subject: User, term: Term) -> Term:
+        """Updates a term.
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            term (Term): Term to update
+
+        Returns:
+            Term: Object updated in the table
+        """
+
+        # Check if user has admin permissions
+        self._permission.enforce(subject, "course.admin", f"term")
+
+        # Find the entity to update
+        term_entity = self._session.get(TermEntity, id)
+
+        # Raise an error if no entity was found
+        if term_entity is None:
+            raise ResourceNotFoundException(f"Term with id: {id} does not exist.")
+
+        # Update the entity
+        term_entity.name = term.name
+        term_entity.start = term.start
+        term_entity.end = term.end
+
+        # Commit changes
+        self._session.commit()
+
+        # Return edited object
+        return term_entity.to_model()
+
+    def delete(self, subject: User, term: Term) -> None:
+        """Deletes a term.
+
+        Parameters:
+            subject: a valid User model representing the currently logged in User
+            term (Term): Term to delete
+        """
+
+        # Check if user has admin permissions
+        self._permission.enforce(subject, "course.admin", f"term")
```

Comment:
Permission `"courses.term.delete"`, `"term/{term_id}"`

This is a bit pedantic because we'll never be giving term level permissions, but for consistency...